### PR TITLE
[mono] More domain cleanups

### PR DESCRIPTION
**Pull request by vargaz**

### CODE REVIEW

**Comment by marek-safar** [comment]
@vargaz could you resolve the confict

@CoffeeFlux @lambdageek plase review

**Review by CoffeeFlux — APPROVED** [review]
LGTM, thanks!

There might end up being conflicts with https://github.com/dotnet/runtime/pull/49637 but they should be easy to resolve on either end.

File: src/mono/mono/metadata/domain.c [hunk 1]
```diff
@@ -212,115 +125,32 @@ gc_free_fixed_non_heap_list (void *ptr)
...
 	MONO_PROFILER_RAISE (domain_loading, (domain));
 
-	domain->domain_assemblies = NULL;
-
-	mono_coop_mutex_init_recursive (&domain->assemblies_lock);
-
-	mono_appdomains_lock ();
-	domain_id_alloc (domain);
-	mono_appdomains_unlock ();
+	// FIXME: why is this needed ?
```
[the remark this answers is not in the document]
Comment (vargaz) [reply src/mono/mono/metadata/domain.c:151]: Will try to remove later.

File: src/mono/mono/metadata/domain.c [hunk 2]
```diff
@@ -212,115 +125,32 @@ gc_free_fixed_non_heap_list (void *ptr)
...
+	MonoGCDescriptor domain_gc_desc;
+
+	unsigned int i, bit = 0;
+	memset (domain_gc_bitmap, 0, sizeof (domain_gc_bitmap));
+	for (i = G_STRUCT_OFFSET (MonoDomain, MONO_DOMAIN_FIRST_OBJECT); i <= G_STRUCT_OFFSET (MonoDomain, MONO_DOMAIN_LAST_OBJECT); i += sizeof (gpointer)) {
+		bit = i / sizeof (gpointer);
+		domain_gc_bitmap [bit / 32] |= (gsize) 1 << (bit % 32);
 	}
-	mono_appdomains_unlock ();
+	domain_gc_desc = mono_gc_make_descr_from_bitmap ((gsize*)domain_gc_bitmap, bit + 1);
```
[the remark this answers is not in the document]
Comment (vargaz) [reply src/mono/mono/metadata/domain.c:142]: Yes, its used to keep the GC refs in MonoDomain alive.

**Pull request merged**